### PR TITLE
Container image streaming blog post

### PR DESCRIPTION
**Pull request by iain-macdonald**
I'd appreciate feedback, especially about whether this is too detailed or not detailed enough.

**Related issues**: N/A

### CODE REVIEW

**Review by siggisim — APPROVED** [review]
This looks great! (modulo the github link missing an https:// prefix)

**Review by tylerwilliams — APPROVED** [review]
looks great!

File: website/blog/container-image-streaming.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+## Cache Eviction
+
+We continuously evict the oldest unused artifacts from the BuildBuddy caches to make room for fresh data, and SOCI artifacts are no exception to this. Fortunately, the average last-access-time (including reads) of evicted items for most customers is old enough that only artifacts for rarely pulled container images are affected. Container images used regularly in remote execution or CI only require re-indexing except after being updated in the container registry.
+
+## Bugs!
+
+After initially rolling out lazy image distribution to our executors, we saw some build and test failures due to `Input/output errors` running binaries in containers occasionally. We tracked this down to [a bug](https://github.com/containerd/containerd/issues/6377) in the containerd docker authorization library that was causing it to re-use expired authentication tokens a bit too eagerly. Fortunately, there was a pretty straightforward [workaround](https://github.com/buildbuddy-io/soci-snapshotter/pull/2) while we wait for the [upstream fix](https://github.com/containerd/containerd/pull/8735/files).
```
Comment (bduffany) [inline website/blog/container-image-streaming.md:58]: idk if you wanted to also add the race condition you found by turning on the race detector - I thought that was really cool!

Comment (iain-macdonald) [reply website/blog/container-image-streaming.md]: How could I forget - thanks for the reminder!

**Review by bduffany — APPROVED** [review]

**Pull request merged**